spec/inputs/http_spec.rb

```diff
@@ -368,6 +297,113 @@
+          let(:config) { { "remote_host_target_field" => "remote_host" } }
+          it "is written to the value of \"remote_host_target_field\" property" do
+            client.post("http://localhost:#{port}/meh.json",
+                        :headers => { "content-type" => "text/plain" },
+                        :body => "hello").call
+            event = logstash_queue.pop
+            expect(event.get("remote_host")).to eq("127.0.0.1")
+          end
+        end
+      end
+
+      describe "request headers" do
+        subject { LogStash::Inputs::Http.new(config.merge("port" => port)) }
+        context "by default" do
+          let(:config) { {} }
+          it "are written to the \"headers\" field" do
+            client.post("http://localhost:#{port}/meh.json",
+                        :headers => { "content-type" => "text/plain" },
+                        :body => "hello").call
+            event = logstash_queue.pop
+            expect(event.get(header_field)).to be_a(Hash)
+            expect(event.get(request_method_field)).to eq("POST")
+            expect(event.get(request_path_field)).to eq("/meh.json")
+            expect(event.get(http_version_field)).to eq("HTTP/1.1")
+            expect(event.get(user_agent_field)).to include("Manticore")
+            if ecs_compatibility == :disabled
+              expect(event.get(http_host_field)).to eq("localhost:#{port}")
+            else
+              expect(event.get(domain_field)).to eq("localhost")
+              expect(event.get(port_field)).to eq("#{port}")
+            end
+
+            expect(event.get(content_length_field)).to eq("5")
+            expect(event.get(content_type_field)).to eq("text/plain")
+          end
+        end
+        context "when using request_headers_target_field" do
+          let(:config) { { "request_headers_target_field" => "request_headers" } }
+          it "are written to the field set in \"request_headers_target_field\"" do
+            client.post("http://localhost:#{port}/meh.json",
+                        :headers => { "content-type" => "text/plain" },
+                        :body => "hello").call
+            event = logstash_queue.pop
+            expect(event.get("request_headers")).to be_a(Hash)
+            expect(event.get("request_headers")).to include("request_method" => "POST")
+            expect(event.get("request_headers")).to include("request_path" => "/meh.json")
+            expect(event.get("request_headers")).to include("http_version" => "HTTP/1.1")
+            expect(event.get("request_headers")["http_user_agent"]).to include("Manticore")
+            expect(event.get("request_headers")).to include("http_host" => "localhost:#{port}")
+            expect(event.get("request_headers")).to include("content_length" => "5")
+            expect(event.get("request_headers")).to include("content_type" => "text/plain")
+          end
+        end
+      end
+    end
+  end
+
+  def setup_client
```

Comment:
The task of `setup_client` is to wait that the http plugin is up and able to respond, so it's not properly a setup of the http client but a wait for server to bootstrap, so rename it with something that expose this behavior is better. Maybe `wait_for_http_is_ready` or something similar

Reply:
I rename it to `setup_server_client`. It is doing a setup for the test, so I am not feeling right to say `wait_xx_ready`. Let me know if it doesn't work for you.